### DISCUSSION TITLE
Question about security plugin.

### DISCUSSION THREAD

**Discussion by advancenXX** [eProsima/Fast-DDS#3180] [Q&A] [answered]
I know we could enable crypto plugin together with authentication plugin, but I want to use crypto plugin without authentication.
I am wondering if it is possible to use my own session key instead of sharedsecret of authentication, by modifying some code.

**Comment by JLBuenoLopez** [comment]
Hi @advancenXX,

Sure, [Fast DDS license](https://github.com/eProsima/Fast-DDS/blob/master/LICENSE) allows user modifications so feel free to modify whatever you want.

**Comment by advancenXX** [comment, marked as answer]
Yes, but it is difficult to modify security plugins.
Disable authentication plugin, then, fill in remote_crypto_handles with my own session key(which is hard to use as remote information should be filled in during discovery phase).
